Assets/Simon/Scripts/GameLoopManager.cs

```diff
@@ -0,0 +1,99 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class GameLoopManager : MonoBehaviour
+{
+    [SerializeField] private Button startGameButton;
+
+    private MiniGamePhases currentMiniGamePhases = MiniGamePhases.WatchingPhase;
+
+    private Sequence sequence;
+
+    private UIManager UImanager;
+
+    private ColorButtonEffectPlayer colorButtonEffectPlayer;
+
+    private int score = 0;
+
+
+    private void Awake()
+    {
+        startGameButton.onClick.RemoveAllListeners();
+        startGameButton.onClick.AddListener(StartGameLoop);
+
+        sequence = GameObject.FindObjectOfType<Sequence>();
+        UImanager = GameObject.FindObjectOfType<UIManager>();
+        colorButtonEffectPlayer = GameObject.FindObjectOfType<ColorButtonEffectPlayer>();
+    }
+
+    private void StartGameLoop()
+    {
+        StartCoroutine(StartGameLoopCoroutine());
+    }
+
+    private IEnumerator StartGameLoopCoroutine()

```

Comment:
this class feels unclean with the amount of IEnumerators, are they all necessary?

Reply:
Actually, the game is split in phases. Before starting, each phase must wait for the previous one to end. This is why I use coroutines.

I don't know if coroutines are the best solution here, I just know that I need sequences to be played one after the other.